File: frontend/erpclink-web/src/app/features/packages/models/package.models.ts
export interface PackageItemDto {
  id: string;
  serviceId: string;
  serviceCode: string;
  serviceName: string;
  quantity: number;
  sortOrder: number;
}

export interface HealthcarePackageDto {
  id: string;
  organizationId: string;
  packageCode: string;
  name: string;
  description: string | null;
  isActive: boolean;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
  rowVersion: string;
  items: PackageItemDto[];
}

export interface HealthcarePackageListItemDto {
  id: string;
  packageCode: string;
  name: string;
  isActive: boolean;
  itemCount: number;
}

export interface PagedHealthcarePackagesResult {
  items: HealthcarePackageListItemDto[];
  page: number;
  pageSize: number;
  totalCount: number;
}

export interface CreateHealthcarePackageRequest {
  packageCode: string;
  name: string;
  description?: string | null;
  items?: { serviceId: string; quantity: number; sortOrder?: number | null }[] | null;
}

export interface UpdateHealthcarePackageRequest {
  name: string;
  description?: string | null;
  rowVersion?: string | null;
}

export interface PackageItemInput {
  serviceId: string;
  quantity: number;
  sortOrder?: number | null;
}

export interface UpdatePackageItemRequest {
  quantity: number;
  sortOrder: number;
  rowVersion?: string | null;
}

export interface SearchHealthcarePackagesParams {
  query?: string;
  packageCode?: string;
  isActive?: boolean;
  page?: number;
  pageSize?: number;
}

File: src/Modules/Services/ErpClink.Modules.Services.Application/Packages/Models/PackageModels.cs
namespace ErpClink.Modules.Services.Application.Packages.Models;

public sealed record CreateHealthcarePackageRequest(
    string PackageCode,
    string Name,
    string? Description,
    IReadOnlyList<PackageItemInput>? Items);

public sealed record UpdateHealthcarePackageRequest(string Name, string? Description, byte[]? RowVersion);

public sealed record PackageItemInput(Guid ServiceId, decimal Quantity, int? SortOrder);

public sealed record UpdatePackageItemRequest(decimal Quantity, int SortOrder, byte[]? RowVersion);

public sealed record SearchHealthcarePackagesRequest(
    string? Query,
    string? PackageCode,
    bool? IsActive,
    int Page = 1,
    int PageSize = 20);

public sealed record PackageItemDto(
    Guid Id,
    Guid ServiceId,
    string ServiceCode,
    string ServiceName,
    decimal Quantity,
    int SortOrder);

public sealed record HealthcarePackageDto(
    Guid Id,
    Guid OrganizationId,
    string PackageCode,
    string Name,
    string? Description,
    bool IsActive,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy,
    byte[] RowVersion,
    IReadOnlyList<PackageItemDto> Items);

public sealed record HealthcarePackageListItemDto(
    Guid Id,
    string PackageCode,
    string Name,
    bool IsActive,
    int ItemCount);

public sealed record PagedHealthcarePackagesResult(
    IReadOnlyList<HealthcarePackageListItemDto> Items,
    int Page,
    int PageSize,
    int TotalCount);
